gulpfile.js

```diff
@@ -92,7 +107,7 @@ var frontendDependencies = manifest.dependencies.frontend;
 var cssFilesPath = [];
 var jsFilesPath = [];
 var fontFolderPath = [];
-var cssBackgroundPath = [];
+var cssBackgroundFilepaths = [];
 var generatedTargetDir = path.join(
   'third_party', 'generated',
   isMinificationNeeded ? 'prod' : 'dev');
```

Comment:
I would say, if something is used only by one task, create it in the task, otherwise create a common function that the tasks call. Even if it's the former case, if there's common functionality between the three loops it might be possible to refactor so that they all make use of a common utility function.